File: android_application/app/src/main/java/org/techtown/smartkey_app/lock.java
package org.techtown.smartkey_app;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ImageButton;
import android.widget.Spinner;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class lock extends AppCompatActivity {

    TextView textView;

    String[] items = { }; //등록하면 추가되도록 해야함

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_lock);

        textView = findViewById(R.id.textView4);

        Spinner spinner = findViewById(R.id.spinner);
        ArrayAdapter<String> adapter = new ArrayAdapter<String> (
                this, android.R.layout.simple_spinner_item, items);
        adapter.setDropDownViewResource(
                android.R.layout.simple_spinner_dropdown_item);

        spinner.setAdapter(adapter);

        spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> adapterView, View view, int position, long id) {
                textView.setText(items[position]);
            }

            @Override
            public void onNothingSelected(AdapterView<?> adapterView) {
                textView.setText("");
            }
        });

        ImageButton imageButton = (ImageButton) findViewById(R.id.plus_btn);
        imageButton.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), add_category.class);
                startActivity(intent);
            }
        });

        ImageButton imageButton2 = (ImageButton) findViewById(R.id.lock_btn);
        imageButton2.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), qr_check.class);
                startActivity(intent);
            }
        });

    }
}

File: smartKey/app/src/main/java/com/zartre/app/beacondemo/lock.kt
package com.zartre.app.beacondemo

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_lock.*

class lock : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_lock)


        plus_btn.setOnClickListener {
            var intent = Intent(applicationContext, add_category::class.java)
            startActivity(intent)
        }

        lock_btn.setOnClickListener {

//          * 소상공인 (사업자)
            var intent = Intent(applicationContext, unlock::class.java)
            startActivity(intent)
//          * 게스트
//              var intent = Intent(applicationContext, qr_check::class.java)
//              startActivity(intent)

        }
    }
}



// 1. 비콘 스캔
// 2. 비콘 스캔한 id랑 서버에서 받은 비콘 id 비교
// 2-1. 같으면 서버에 "True, 비콘 id" 전송
// 2-2. 다르면 서버에 보내는거 없이 dialog msg만 띄우기